rust/ql/consistency-queries/CfgConsistency.ql

```diff
@@ -25,3 +25,9 @@ query predicate scopeNoFirst(CfgScope scope) {
   not scope = any(Function f | not exists(f.getBody())) and
   not scope = any(ClosureExpr c | not exists(c.getBody()))
 }
+
+query predicate deadEnd(CfgImpl::Node node) {
+  Consistency::deadEnd(node) and
+  // `else` blocks in `let` statements diverge, so they are by definition dead ends
+  not node.getAstNode() = any(LetStmt let).getLetElse().getBlockExpr()
```

Comment:
I think this is not right. The `else` branch should complete abnormally (`panic`), or with something like a `return`, `break` or `continue`.  The following should be fine
```rust
        loop {
            let Some(_x) = iter.next() else {
                break;
            };
        }
```

Reply:
Right; I think that is actually handled in the CFG implementation; I'll add a test case.